allow to reference files and folders for platforms with local requires

### I tried this:

use a platform folder / file which references other files, e.g. something like this https://github.com/winglang/wing-cloud-platform/tree/main/examples

```
 wing compile -t ../lib simple.main.w
```

### This happened:

```
An error occurred while loading the custom platform: node:internal/modules/cjs/loader:1080
  throw err;
  ^

Error: Cannot find module './test-platform'
Require stack:
```

### I expected this:

it works without errors

### Is there a workaround?

build a local npm package for the platform (`npm pack`) und install it in the example for usage

### Anything else?

_No response_

### Wing Version

0.48

### Node.js Version

18.7

### Platform(s)

MacOS

### Community Notes

<!-- Please keep this note for the community -->
* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

--- a/examples/tests/valid/platforms/local-package/a-relative-platform.js
+++ b/examples/tests/valid/platforms/local-package/a-relative-platform.js
@@ -1,0 +1,4 @@
+exports.MyPlatform = class MyPlatform {
+  target = "tf-aws";
+}
+

--- a/examples/tests/valid/platforms/local-package/index.js
+++ b/examples/tests/valid/platforms/local-package/index.js
@@ -1,0 +1,2 @@
+const { MyPlatform } = require("./a-relative-platform");
+exports.Platform = MyPlatform;

--- a/libs/wingsdk/src/platform/platform-manager.ts
+++ b/libs/wingsdk/src/platform/platform-manager.ts
@@ -136,11 +136,17 @@ export function _loadCustomPlatform(customPlatformPath: string): any {
     ? join(platformDir, `${customPlatformPath}/lib/index.js`)
     : `${customPlatformPath}/index.js`;
 
+  // enable relative imports from the platform file
+  const customPlatformBaseDir = customPlatformPath.endsWith(".js")
+    ? dirname(customPlatformPath)
+    : customPlatformPath;
+
   const cwdNodeModules = join(cwd(), "node_modules");
   const customPlatformLib = join(cwdNodeModules, customPlatformPath, "lib");
 
   const resolvablePaths = [
     ...module.paths,
+    customPlatformBaseDir,
     platformDir,
     cwdNodeModules,
     customPlatformLib,

--- a/tools/hangar/src/platform.test.ts
+++ b/tools/hangar/src/platform.test.ts
@@ -60,6 +60,33 @@ describe("Platform examples", () => {
 
   });
 
+  describe("Platform folder with relative dependencies", () => {
+    const args = ["compile"];
+    const basePlatforms = ["tf-aws"]
+
+    test("compile local package from folder", async () => {
+      const platform = path.join(platformsDir, "local-package");
+      await runWingCommand({
+        cwd: tmpDir,
+        platforms: [...basePlatforms, platform],
+        wingFile: appFile,
+        args,
+        expectFailure: false
+      });
+    })
+
+    test("compile local package from file", async () => {
+      const platform = path.join(platformsDir, "local-package", "index.js");
+      await runWingCommand({
+        cwd: tmpDir,
+        platforms: [...basePlatforms, platform],
+        wingFile: appFile,
+        args,
+        expectFailure: false
+      });
+    })
+  });
+
   describe("AWS target platform", () => {
     const args = ["compile"];
     const basePlatforms = ["tf-aws"]
@@ -67,7 +94,7 @@ describe("Platform examples", () => {
 
     test("permission-boundary.js", async () => {
       const platform = path.join(platformsDir, "permission-boundary.js");
-  
+
       const expectedPermissionBoundaryArn = "some:fake:arn:SUPERADMIN";
       process.env.PERMISSION_BOUNDARY_ARN = expectedPermissionBoundaryArn;
       await runWingCommand({
@@ -171,7 +198,7 @@ describe("Platform examples", () => {
 
       test("gcp backend", async () => {
         const platform = path.join(platformsDir, tfBackendPluginName);
-        
+
         const tfBackend = "gcs";
         const tfBackendBucket = "my-wing-bucket";
         const stateFile = "some-state-file.tfstate";

